### ISSUE TITLE
Rule Proposal: Ensure prop.type is an Object

### ISSUE THREAD

**Issue by richardtallent** [vuejs/eslint-plugin-vue#323]
**Please describe what the rule should do:**

Ensure that the `type` key for a property is passed as an Object, not as a string.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Enforces code style
[x] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
props: {
	myFieldWithBadType: {
		type: "Object",
		default: function() {
			return {}
		},
	},
	myOtherFieldWithBadType: {
		type: "Number",
		default: 1,
	},
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**

Passing a string to `type` rather than the name of the type results in a rather opaque error message from deep in the bowels of Vue, and it's an easy mistake to make.

**Comment by michalsnik** [comment]
This is an interesting proposal @richardtallent It might indeed be useful for newcomers that learn Vue 👍

**Comment by chrisvfritz** [comment]
I like this! It would go in `essentials` and a more complete list of situations to match might be:

``` js
import Foo from './foo'
const Bar = ''

// ...

props: {
  a: 'Object',
  b: { type: 'Number' },
  c: { type: `String` },
  d: { type: Foo + '' },
  e: { type: Bar },
}
```
